### PR TITLE
VoyageAI reranking bug fix

### PR DESCRIPTION
**Pull request by fzowl**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new unit/integration tests
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/postprocessor/llama-index-postprocessor-voyageai-rerank/llama_index/postprocessor/voyageai_rerank/base.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def __init__(
             )
 
         self._client = Client(api_key=api_key)
-        super().__init__(top_n=top_k, model=model, truncation=truncation)
+        super().__init__(top_k=top_k, model=model, truncation=truncation)
```
Comment (logan-markewich) [inline llama-index-integrations/postprocessor/llama-index-postprocessor-voyageai-rerank/llama_index/postprocessor/voyageai_rerank/base.py:42]: actually, now that I'm seeing this, I'm pretty sure all our other rerankers use the `top_n` kwarg 👁️

**Review by logan-markewich — COMMENTED** [review]
fyi, need to merge main to fix cicd for now

File: llama-index-integrations/postprocessor/llama-index-postprocessor-voyageai-rerank/llama_index/postprocessor/voyageai_rerank/base.py (hunk 1 above)
Comment (fzowl) [reply llama-index-integrations/postprocessor/llama-index-postprocessor-voyageai-rerank/llama_index/postprocessor/voyageai_rerank/base.py]: I changed to top_n

**Comment by fzowl** [comment]
> fyi, need to merge main to fix cicd for now

@logan-markewich Hm, I tried to add `pytest-mock` as a top-level dev dependency, but it did not solve the problem. Do you have an idea how I can fix the `mocker` issue here? I'm okay with commenting out the test, but I have mixed feelings: I feel that this unit test would be great to have, but on the other hand, I now know that the integration works, and there are no similar unit tests in the other reranker integrations.

**Comment by logan-markewich** [comment]
@fzowl pants (our cicd tools) pulls in dependencies based on imports

You should import and type the function

```
from pytest_mock import MockerFixture

...

def test_rerank(mocker: MockerFixture) -> None:
  ...
```

**Comment by fzowl** [comment]
@logan-markewich tests are successful now; thank you for the explanation!

File: llama-index-integrations/postprocessor/llama-index-postprocessor-voyageai-rerank/tests/test_postprocessor_voyageai-rerank.py [hunk 2]
```diff
@@ -1,7 +1,42 @@
...
+        response=VoyageResponse.construct_from(rerank_sample_response),
+    )
+
+    # Mock create_client to return our mock_client
+    mocker.patch.object(VoyageAIRerank, "_client", return_value=mock_client)
+
+    voyageai_rerank = VoyageAIRerank(
+        api_key="api_key", top_n=2, model="rerank-lite-1", truncation=True
+    )
+    voyageai_rerank._postprocess_nodes(
```
Comment (nerdai) [inline llama-index-integrations/postprocessor/llama-index-postprocessor-voyageai-rerank/tests/test_postprocessor_voyageai-rerank.py:39]: nit: should use `postprocess_nodes()` instead.

also should we be asserting something here for it to be a good test?

**Review by nerdai — COMMENTED** [review]
thanks @fzowl for the fix. Is there a link to the bug/issue that you're fixing here?

File: llama-index-integrations/postprocessor/llama-index-postprocessor-voyageai-rerank/tests/test_postprocessor_voyageai-rerank.py (hunk 2 above)
Comment (fzowl) [reply llama-index-integrations/postprocessor/llama-index-postprocessor-voyageai-rerank/tests/test_postprocessor_voyageai-rerank.py]: Made the changes here.

**Comment by fzowl** [comment]
@nerdai I don't know if there is any reported bug/issue. I was double checking our (VoyageAI) integrations and during this exercise i run into the reranking implementation here.

**Comment by nerdai** [comment]
@fzowl got it. Thanks!

**Review by nerdai — APPROVED** [review]
Lgtm!

**Pull request merged**